agent/php_internal_instrument.c

```diff
@@ -1531,6 +1533,71 @@ NR_INNER_WRAPPER(memcache_function) {
                                              INTERNAL_FUNCTION_PARAM_PASSTHRU);
 }
 
+NR_INNER_WRAPPER(memcached_add_server) {
+  char* host = NULL;
+  nr_string_len_t host_len = 0;
+  zend_long port = 0;
+  zend_long weight = 0;
+  nr_datastore_instance_t* instance = NULL;
+  char* instance_metric = NULL;
+  int zcaught = 0;
+
+  if (SUCCESS
+      == zend_parse_parameters_ex(
+          ZEND_PARSE_PARAMS_QUIET, ZEND_NUM_ARGS(), "s|ll", &host,
+          &host_len, &port, &weight) &&
+      NULL != host) {
+    instance = nr_php_memcached_create_datastore_instance(host, port);
+    instance_metric = nr_formatf("Datastore/instance/Memcached/%s/%s",
+                               instance->host, instance->port_path_or_id);
+    nrm_force_add(NRPRG(txn)->unscoped_metrics, instance_metric, 0);
+    nr_datastore_instance_destroy(&instance);
+    nr_free(instance_metric);
+  }
+  zcaught = nr_zend_call_old_handler(nr_wrapper->oldhandler,
+                                     INTERNAL_FUNCTION_PARAM_PASSTHRU);
+  if (zcaught) {
+    zend_bailout();
+    /* NOTREACHED */
+  }
+}
+
+NR_INNER_WRAPPER(memcached_add_servers) {
+  zval* servers = NULL;
+  zval* server = NULL;
+  nr_datastore_instance_t* instance = NULL;
+  char* instance_metric = NULL;
+  int zcaught = 0;
+
+  if (SUCCESS
+      == zend_parse_parameters_ex(
+          ZEND_PARSE_PARAMS_QUIET, ZEND_NUM_ARGS(), "a", &servers)) {
+    if (Z_TYPE_P(servers) == IS_ARRAY) {
```

Comment:
Ensure that `servers` is not `NULL` before dereferencing it in `Z_TYPE_P` macro:
```suggestion
    if (NULL != servers && Z_TYPE_P(servers) == IS_ARRAY) {
```